modules/038-registry/hooks/checker/build_queue.go

```diff
@@ -18,46 +18,54 @@ package checker
 
 import (
 	"fmt"
-	"maps"
 	"sort"
-	"strings"
 
 	gcr_name "github.com/google/go-containerregistry/pkg/name"
-)
 
-func buildRepoQueue(info clusterImagesInfo, repo gcr_name.Repository) ([]queueItem, error) {
-	currentRepo, err := gcr_name.NewRepository(info.Repo)
-	if err != nil {
-		return nil, fmt.Errorf("cannot parse registry base %q: %w", info.Repo, err)
-	}
+	registry_const "github.com/deckhouse/deckhouse/go_lib/registry/const"
+)
 
+func buildRepoQueue(info clusterImagesInfo, repo gcr_name.Repository, checkMode registry_const.CheckModeType) ([]queueItem, error) {
 	images := make(map[string]string)
-	for d, info := range info.ModulesImagesDigests {
-		image := currentRepo.Digest(d).String()
-		images[image] = info
-	}
-
-	// Merge deckhouse images to modules images
-	maps.Copy(images, collectDeckhouseQueueImages(info.DeckhouseImages, currentRepo))
 
-	ret := make([]queueItem, 0, len(images))
-	for image, info := range images {
-		ref, err := gcr_name.ParseReference(image)
+	switch checkMode {
+	case registry_const.Relax:
+		// Only deckhouse container image
+		newImageRef, err := updateImageRepo(info.DeckhouseContainerImageRef, repo)
 		if err != nil {
-			return ret, fmt.Errorf("cannot parse image %q (%q) reference: %w", image, info, err)
+			return nil, fmt.Errorf("failed to update image reference: %w", err)
 		}
 
-		if !strings.HasPrefix(ref.String(), currentRepo.String()) {
-			return ret, fmt.Errorf("image %q (%q) ref not starts with repository %q", ref.String(), info, currentRepo.String())
+		// Need to be a tag
+		if _, ok := newImageRef.(gcr_name.Tag); !ok {
+			return nil, fmt.Errorf("expected deckhouse image reference to be a tag, but got: %s", newImageRef.String())
+		}
+		images[newImageRef.String()] = "deckhouse/containers/deckhouse"
+	default:
+		images = make(map[string]string, len(info.ModulesImagesDigests)+len(info.DeckhouseImagesRefs))
+
+		// Module images
+		for digest, info := range info.ModulesImagesDigests {
+			image := repo.Digest(digest).String()
+			images[image] = info
 		}
 
-		imagePath := strings.TrimPrefix(ref.String(), currentRepo.String())
+		// Deckhouse images
+		for info, image := range info.DeckhouseImagesRefs {
+			newImageRef, err := updateImageRepo(image, repo)
+			if err != nil {
+				return nil, err
+			}
+			images[newImageRef.String()] = info
+		}
+	}
 
+	ret := make([]queueItem, 0, len(images))
```

Comment:
duplicate make of `images` map, please use `var` on top of function and `make` in each `case` block